src/App.jsx

```diff
@@ -14,17 +15,32 @@ function App() {
     { id: "143", invoice_id: "testinvoice" },
   ]);
 
-  useEffect(() => {
-    const fetchData = async () => {
-      const response = await fetch(INVOICE_URL);
-      const json = await response.json();
-      setInvoices(json.data);
-      setLoading(false);
-    };
+  const fetchAllData = async () => {
+    const response = await fetch(INVOICE_URL);
+    const json = await response.json();
+    setInvoices(json.data);
+    setLoading(false);
+  };
 
-    fetchData().catch(console.error);
+  useEffect(() => {
+    fetchAllData().catch(console.error);
   }, []);
 
+  const fetchFilteredData = async (status) => {
+    const response = await fetch(INVOICE_URL + "?status=" + status);
+    const json = await response.json();
+    setInvoices(json.data);
+    setLoading(false);
+  };
```

Comment:
You could be a bit more DRY by having just one fetcher with an optional `status` parameter that fetches everything when called without it.